### PR TITLE
powershell_script: Use native properties and don't coerce user provided flags with powershell defaults

### PR DESCRIPTION
**Pull request by phiggins**
Some code I had sitting in a local branch. If I remember right this was part of a bigger change that I abandoned and I forgot to PR the remainder.

### CODE REVIEW

File: lib/chef/resource/execute.rb [hunk 1]
```diff
@@ -509,7 +509,6 @@ class Execute < Chef::Resource
       def initialize(name, run_context = nil)
         super
         @command = name
-        @backup = 5
```
Comment (tas50) [inline lib/chef/resource/execute.rb:512]: where did this default go?

Comment (tas50) [reply lib/chef/resource/execute.rb:512]: I bet this is leftover from when backup wasn't a property

File: spec/unit/resource/powershell_script_spec.rb [hunk 2]
```diff
@@ -130,24 +130,7 @@
...
-    before do
-      allow(@resource).to receive(:default_flags).and_return(nil)
-    end
-    it_behaves_like "a Windows script resource"
-  end
-
-  describe "#flags" do
-    let(:resource) { @resource }
 
-    it "appends user's flags to the defaults" do
```
Comment (tas50) [inline spec/unit/resource/powershell_script_spec.rb:142]: This was a particular bug that we wanted to make sure we didn't introduce a regression for. If someone has an option that was basically foo false and default set it to true we want to make sure we're always applying the user supplied values last. We need some sort of test to make sure refactoring later doesn't break the order.

Comment (phiggins) [reply spec/unit/resource/powershell_script_spec.rb:142]: Good call. I can probably add a test for the command output.

File: lib/chef/resource/execute.rb (hunk 1 above)
Comment (phiggins) [reply lib/chef/resource/execute.rb:512]: It's probably this: https://github.com/chef/chef/blob/master/lib/chef/resource/file.rb#L65-L66

**Pull request merged**